flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-paimon/src/main/java/org/apache/flink/cdc/connectors/paimon/sink/v2/PreCommitOperator.java

```diff
@@ -140,4 +147,9 @@ public void close() throws Exception {
             storeMultiCommitter.close();
         }
     }
+
+    @Override
+    public void endInput() throws Exception {
+        commitUpToCheckpoint(Long.MAX_VALUE);
+    }
```

Comment:
When using flink batch mode and bounded stream, the endInput method of BoundedOneInput is needed to drive commit.
In the test case, env.fromCollection(events, TypeInformation.of(Event.class) returns a bounded stream. When the data ends, endInput() is called to generate a paimon snapshot.